oga/backend/users/views.py

```diff
@@ -1,42 +1,68 @@
 from django.shortcuts import render, get_object_or_404
-from django.http import HttpResponse, HttpResponseBadRequest, JsonResponse
+from django.http import HttpResponse, HttpResponseBadRequest, JsonResponse, HttpResponseNotAllowed
 from django.views.decorators.csrf import csrf_exempt
 import json
 from .models import Profile, Question, Answer, Location
 from django.contrib.auth.models import User
 from django.contrib.auth import login, authenticate, get_user
 from django.views import generic
+from functools import wraps
 
+def login_required(views_func):
+    @wraps(views_func)
+    def wrapper(request, *args, **kwargs):
+        if request.user.is_authenticated:
+            return views_func(request, *args, **kwargs)
+        else:
+            return HttpResponse(status=400)
+    return wrapper
+
+def check_request(views_func):
+    @wraps(views_func)
+    def wrapper(request, *args, **kwargs):
+        try:
+            return views_func
+        except (KeyError, ValueError) as e:
+            return HttpResponse(status=400)
```

Comment:
I think we don't have to specify request argument here, since we don't use it unlike login_required.

```suggestion
    def inner(*args, **kwargs):
        try:
            return view_func(*args, **kwargs)
        except (KeyError, json.JSONDecodeError) as e:
            return HttpResponseBadRequest(str(e))
```

perhaps something like this would be better..?

Also, as of now, because we only return views_func (instead of ```views_func(request, *args, **kwargs) ```executing this line causes error along the lines of ```AttributeError: 'function' object has no attribute 'get'```

Reply:
Great. I follow your suggestions.